addons/payment/models/payment_acquirer.py

```diff
@@ -200,17 +239,40 @@ def _get_feature_support(self):
     @api.model
     def create(self, vals):
         image_resize_images(vals)
+        if not vals.get('display'):
+            payment_icon_ids = vals.get('payment_icon_ids')
+            if payment_icon_ids and len(payment_icon_ids[0]) >= 3 and payment_icon_ids[0][2]:
+                vals['display'] = "Credit Card (powered by {0})".format(vals['name'])
+            else:
+                vals['display'] = vals['name']
         return super(PaymentAcquirer, self).create(vals)
 
     @api.multi
     def write(self, vals):
         image_resize_images(vals)
-        return super(PaymentAcquirer, self).write(vals)
+        super(PaymentAcquirer, self).write(vals)
+        for acquirer in self:
+            vals = {}
+            if acquirer.missing_required_fields:
+                vals['website_published'] = False
+                super(PaymentAcquirer, self).write(vals)
```

Comment:
You don't return any result here ! And it looks strange to erase the vals dictionary. Why don't you just do something like:
```python
    @api.multi
    def write(self, vals):
        image_resize_images(vals)
        res = super(PaymentAcquirer, self).write(vals)
        missing_fields_acquirers = self.filtered(lambda a: a.missing_required_fields)
        super(PaymentAcquirer, missing_fields_acquirers).write({'website_published': False})
        return res
```